java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java

```diff
@@ -57,6 +58,11 @@ public void scanFile(JavaFileScannerContext context) {
   }
 
   @Override
+  public void visitEnumConstant(EnumConstantTree tree) {
+    // Ignore literals within enum constant
```

Comment:
Do you see a case where the initializer of the enum could be something else than a NewClassTree ?